src/Stratis.Bitcoin.Features.PoA/Voting/PollsRepository.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using DBreeze;
+using DBreeze.DataTypes;
+using DBreeze.Utils;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.PoA.Voting
+{
+    public class PollsRepository : IDisposable
+    {
+        private readonly DBreezeEngine dbreeze;
+
+        private readonly ILogger logger;
+
+        private readonly DBreezeSerializer dBreezeSerializer;
+
+        internal const string TableName = "DataTable";
+
+        private static readonly byte[] RepositoryHighestIndexKey = new byte[0];
+
+        private int highestPollId;
+
+        public PollsRepository(DataFolder dataFolder, ILoggerFactory loggerFactory, DBreezeSerializer dBreezeSerializer)
+            : this(dataFolder.PollsPath, loggerFactory, dBreezeSerializer)
+        {
+        }
+
+        public PollsRepository(string folder, ILoggerFactory loggerFactory, DBreezeSerializer dBreezeSerializer)
+        {
+            Guard.NotEmpty(folder, nameof(folder));
+
+            Directory.CreateDirectory(folder);
+            this.dbreeze = new DBreezeEngine(folder);
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.dBreezeSerializer = dBreezeSerializer;
+        }
+
+        public void Initialize()
+        {
+            // Load highest index.
+            this.highestPollId = -1;
+
+            using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
+            {
+                Row<byte[], int> row = transaction.Select<byte[], int>(TableName, RepositoryHighestIndexKey);
+
+                if (row.Exists)
+                    this.highestPollId = row.Value;
```

Comment:
What about reorgs?

Reply:
>What about reorgs?

It will remove items in the repo

>if node craches and the `highestPollId` is out of sync?

Same as with most other components- we won't recover and will have to resync. 
After introduction of tips manager we might fix it

Reply:
Most other components will recover, by reorging the node.